### PR TITLE
addpatch: juce, ver=7.0.12-1

### PR DESCRIPTION
**Pull request by YHStar**
- Add loongarch64 support in *JUCE platform detect* and `cpu_relax()` *func*
- Use *dos2unix* to solve chaos CRLF to LF , to apply patch

### CODE REVIEW

File: elephantdsp-roomreverb/loong.patch [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
++  patch -p1 -i "$srcdir/juce_platform.patch"
+ }
+ 
+ build() {
+@@ -130,3 +136,8 @@ package_elephantdsp-roomreverb-vst3() {
+   install -vd "$pkgdir/usr/lib/vst3"
+   cp -vr VST3/RoomReverb.vst3 "$pkgdir/usr/lib/vst3"
+ }
++
++source+=('github.com-JUCE::git+https://github.com/juce-framework/JUCE'
```
Comment (wszqkzqk) [inline elephantdsp-roomreverb/loong.patch:25]: Please fix on a certain commit to provide reproducible builds.

**Review by wszqkzqk — CHANGES_REQUESTED** [review]

File: elephantdsp-roomreverb/loong.patch (hunk 1 above)
Comment (YHStar) [reply elephantdsp-roomreverb/loong.patch:25]: Original PKGBUILD from ArchLinux all use the newest main branch.

File: juce/loong_fix_relax.patch [hunk 2]
```diff
@@ -0,0 +1,11 @@
+--- a/modules/juce_audio_devices/native/oboe/include/oboe/StabilizedCallback.h	2024-04-15 17:46:51.000000000 +0800
++++ b/modules/juce_audio_devices/native/oboe/include/oboe/StabilizedCallback.h	2024-11-27 19:40:34.884000000 +0800
+@@ -60,7 +60,7 @@
+ #if defined(__i386__) || defined(__x86_64__)
+ #define cpu_relax() asm volatile("rep; nop" ::: "memory");
+ 
+-#elif defined(__arm__) || defined(__mips__) || defined(__riscv)
++#elif defined(__arm__) || defined(__mips__) || defined(__riscv) || defined(__loongarch64)
```
Comment (wszqkzqk) [inline juce/loong_fix_relax.patch:8]: `__loongarch64` is deprecated. Read [this](https://wszqkzqk.github.io/2024/08/22/loongarchlinux-port-tips/#%E6%9C%89%E5%85%B3loongarch64%E7%9A%84%E7%BC%96%E8%AF%91%E5%99%A8%E9%A2%84%E5%AE%9A%E4%B9%89%E5%AE%8F).

**Review by wszqkzqk — CHANGES_REQUESTED** [review]

File: juce/loong_fix_relax.patch (hunk 2 above)
Comment (YHStar) [reply juce/loong_fix_relax.patch]: adjusted

File: juce/loong_fix_relax.patch [hunk 3]
```diff
@@ -0,0 +1,11 @@
+--- a/modules/juce_audio_devices/native/oboe/include/oboe/StabilizedCallback.h	2024-04-15 17:46:51.000000000 +0800
++++ b/modules/juce_audio_devices/native/oboe/include/oboe/StabilizedCallback.h	2024-11-27 19:40:34.884000000 +0800
+@@ -60,7 +60,7 @@
+ #if defined(__i386__) || defined(__x86_64__)
+ #define cpu_relax() asm volatile("rep; nop" ::: "memory");
+ 
+-#elif defined(__arm__) || defined(__mips__) || defined(__riscv)
++#elif defined(__arm__) || defined(__mips__) || defined(__riscv) || defined(__loongarch64_lp64)
```
Comment (wszqkzqk) [inline juce/loong_fix_relax.patch:8]: Other architectures here do not emphasize 64-bit, do we really need to emphasize it?

Comment (YHStar) [reply juce/loong_fix_relax.patch:8]: Arch `aarch64` is also difined here below

`#elif defined(__aarch64__)`
`#define cpu_relax() asm volatile("yield" ::: "memory")`

Comment (wszqkzqk) [reply juce/loong_fix_relax.patch:8]: Why not `__loongarch__`

Comment (YHStar) [reply juce/loong_fix_relax.patch:8]: It's OK, just I have no loongarch32 machine and I didn't know whether the *JUCE* should supports loong32

Comment (wszqkzqk) [reply juce/loong_fix_relax.patch:8]: > It's OK, just I have no loongarch32 machine and I didn't know whether the _JUCE_ should supports loong32

This is actually less important, it's just to fit the style

Comment (YHStar) [reply juce/loong_fix_relax.patch]: adjusted

**Review by wszqkzqk — APPROVED** [review]

**Pull request merged**